app/controllers/api/v1/keywords_controller.rb

```diff
@@ -12,11 +12,38 @@ def index
         render json: KeywordSerializer.new(keywords_list, pagy_options(pagy))
       end
 
+      def create
+        if csv_form.save(create_params[:file])
+          render json: success_create
+
+        else
+          render_errors(
+            details: csv_form.errors.full_messages,
+            code: :invalid_file,
+            status: :unprocessable_entity
+          )
+        end
+      end
+
       private
 
       def keywords
         KeywordsQuery.new(current_user).call
       end
+
+      def csv_form
+        @csv_form ||= CSVUploadForm.new(current_user)
+      end
+
+      def create_params
+        params.permit(:file)
+      end
+
+      def success_create
```

Comment:
Since we are depending on return value of the function, should we rename it to `creation_success_response` or `create_success_response` ?

Reply:
yes, totally better/clearer!! 👌